Significant warnings on R-devel

Rttf2pt1 is raising significant warnings on R-devel, probably because a new compiler warning has been added to the list of **significant** warnings.

https://cran.r-project.org/web/checks/check_results_Rttf2pt1.html

For the record, here are the warnings on r-devel-linux-x86_64-fedora-gcc:

```
checking whether package ‘Rttf2pt1’ can be installed ... [15s/17s] WARNING
Found the following significant warnings:
  t1asm.c:374:51: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘long unsigned int’ [-Wformat=]
  t1asm.c:388:19: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘long int’ [-Wformat=]
  bitmap.c:193:36: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:489:49: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GEX_FRAG *’ {aka ‘struct gex_frag *’} [-Wformat=]
  bitmap.c:489:57: warning: format ‘%p’ expects argument of type ‘void *’, but argument 5 has type ‘GEX_FRAG *’ {aka ‘struct gex_frag *’} [-Wformat=]
  bitmap.c:489:66: warning: format ‘%p’ expects argument of type ‘void *’, but argument 7 has type ‘GEX_FRAG *’ {aka ‘struct gex_frag *’} [-Wformat=]
  ttf2pt1.c:647:40: warning: format ‘%d’ expects argument of type ‘int *’, but argument 3 has type ‘int’ [-Wformat=]
  ttf2pt1.c:647:43: warning: format ‘%d’ expects argument of type ‘int *’, but argument 4 has type ‘int’ [-Wformat=]
  ttf2pt1.c:659:41: warning: format ‘%i’ expects argument of type ‘int *’, but argument 3 has type ‘unsigned int *’ [-Wformat=]
  ttf2pt1.c:688:47: warning: format ‘%i’ expects argument of type ‘int *’, but argument 3 has type ‘unsigned int *’ [-Wformat=]
  ttf2pt1.c:703:67: warning: format ‘%i’ expects argument of type ‘int *’, but argument 3 has type ‘unsigned int *’ [-Wformat=]
  ttf2pt1.c:720:48: warning: format ‘%i’ expects argument of type ‘int *’, but argument 3 has type ‘unsigned int *’ [-Wformat=]
  ttf2pt1.c:1081:36: warning: format ‘%d’ expects argument of type ‘int *’, but argument 3 has type ‘unsigned int *’ [-Wformat=]
  bitmap.c:585:85: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:585:88: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:651:71: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:651:74: warning: format ‘%p’ expects argument of type ‘void *’, but argument 5 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:659:62: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:659:92: warning: format ‘%p’ expects argument of type ‘void *’, but argument 7 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:673:77: warning: format ‘%p’ expects argument of type ‘void *’, but argument 5 has type ‘struct gentry *’ [-Wformat=]
  bitmap.c:678:51: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘struct gentry *’ [-Wformat=]
  bitmap.c:699:74: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:709:61: warning: format ‘%p’ expects argument of type ‘void *’, but argument 5 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:714:35: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘struct gentry *’ [-Wformat=]
  bitmap.c:1101:92: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:1186:41: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:300:81: warning: format ‘%p’ expects argument of type ‘void *’, but argument 5 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:306:66: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:306:79: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:306:92: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 5 has type ‘struct gentry *’ [-Wformat=]
  pt1.c:317:67: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:325:74: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:325:87: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 4 has type ‘struct gentry *’ [-Wformat=]
  pt1.c:325:100: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 5 has type ‘struct gentry *’ [-Wformat=]
  pt1.c:333:83: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:333:96: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 4 has type ‘struct gentry *’ [-Wformat=]
  pt1.c:333:109: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 5 has type ‘struct gentry *’ [-Wformat=]
  pt1.c:341:73: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:341:85: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:1215:41: warning: format ‘%p’ expects argument of type ‘void *’, but argument 5 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:341:98: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 5 has type ‘struct gentry *’ [-Wformat=]
  bitmap.c:1186:41: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘struct gentry *’ [-Wformat=]
  pt1.c:763:51: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:913:50: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:1768:109: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:1768:112: warning: format ‘%p’ expects argument of type ‘void *’, but argument 5 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:1816:70: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:1857:114: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:1866:114: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘struct gentry *’ [-Wformat=]
  bitmap.c:1937:100: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:1988:88: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:1999:106: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:1043:51: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:2038:106: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:1105:51: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:2049:106: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:2071:106: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:1203:40: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:2080:112: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:2080:115: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:2122:102: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:2122:115: warning: format ‘%p’ expects argument of type ‘void *’, but argument 6 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:2131:102: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:2137:86: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:2144:104: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:2144:107: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:2173:101: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:1336:50: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:2184:101: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:1364:50: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:2362:76: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:2482:78: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:2482:81: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  bitmap.c:2482:100: warning: format ‘%p’ expects argument of type ‘void *’, but argument 7 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:2920:97: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 5 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:2920:105: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 6 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:3912:115: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:3912:120: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:3927:111: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:3927:116: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:4058:56: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:4079:57: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:4130:59: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:4152:64: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:4692:77: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:4692:81: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 5 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:4692:85: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 6 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:5867:56: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:5883:82: warning: format ‘%p’ expects argument of type ‘void *’, but argument 5 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:5906:46: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:5906:59: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:5923:103: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:5936:53: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:5945:47: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:5945:61: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:5961:114: warning: format ‘%p’ expects argument of type ‘void *’, but argument 5 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:5975:53: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:5979:47: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:6081:80: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:6081:83: warning: format ‘%p’ expects argument of type ‘void *’, but argument 5 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:6148:61: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:6148:67: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:6176:61: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:6176:64: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:6182:54: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:6182:64: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 5 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:6186:66: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:6186:71: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:6212:87: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:6219:95: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:6224:54: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:6240:87: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘struct gentry *’ [-Wformat=]
  pt1.c:6247:54: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:6255:63: warning: format ‘%p’ expects argument of type ‘void *’, but argument 4 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:6255:66: warning: format ‘%p’ expects argument of type ‘void *’, but argument 5 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:7071:66: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 3 has type ‘GENTRY *’ {aka ‘struct gentry *’} [-Wformat=]
  pt1.c:7071:74: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 4 has type ‘struct gentry *’ [-Wformat=]
  pt1.c:7071:80: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 5 has type ‘struct gentry *’ [-Wformat=]
See https://www.r-project.org/nosvn/R.check/r-devel-linux-x86_64-fedora-gcc/Rttf2pt1-00install.html for details.
```

--- a/NEWS.md
+++ b/NEWS.md
@@ -1,3 +1,8 @@
+Version 1.3.10.9000
+-------------
+
+* Closed #15: fixed `-Wformat` and `-Wformat-pedantic` warnings, which raise significant warnings on R-devel. (#16)
+
 Version 1.3.10
 -------------
 

--- a/src/ttf2pt1/bitmap.c
+++ b/src/ttf2pt1/bitmap.c
@@ -190,7 +190,7 @@ gex_dump_contour(
 		fprintf(stderr, "%-8s", gxf_name[j]);
 		for(i = 0; i < clen; i++, ge = ge->frwd)
 			fprintf(stderr, " %2d", X_FRAG(ge)->len[j]);
-		fprintf(stderr, " %p\n (back) ", ge);
+		fprintf(stderr, " %p\n (back) ", (void*)ge);
 		for(i = 0; i < clen; i++, ge = ge->frwd)
 			fprintf(stderr, " %2d", X_FRAG(ge)->lenback[j]);
 		fprintf(stderr, "\n");
@@ -487,7 +487,7 @@ dosubfrag(
 			xf = lf; /* the symmetric fragment */
 		}
 		fprintf(stderr, "     sym with %p f=%d(%p) e=%d(%p) a1=%c a2=%c sympt=%g\n",
-			xf, symfront, pf, symend, lf,
+			(void*)xf, symfront, (void*)pf, symend, (void*)lf,
 			a1 ? 'Y': 'X', a2 ? 'Y': 'X', sympt
 		);
 		for(i=0; i<4; i++) {
@@ -583,7 +583,7 @@ dosubfrag(
 		ndots--;
 	if( fcrossrayscv(f->vect, NULL, NULL) == 0) {
 		fprintf(stderr, "**** Internal error: rays must cross but don't at %p-%p\n",
-			ge, gel);
+			(void*)ge, (void*)gel);
 		fprintf(stderr, "  (%g, %g) (%g, %g) (%g, %g) (%g, %g)\n", 
 			limfront[0], limfront[1],
 			fixfront[0], fixfront[1],
@@ -649,7 +649,7 @@ frag_subtract(
 			for(pge = ge->frwd, j = slen-1; j > 0; pge = pge->frwd, j--)
 				X_FRAG(pge)->lenback[d] = 0;
 			fprintf(stderr, "    cut %s circular frag to %p-%p\n", 
-				gxf_name[d], pge, ge);
+				gxf_name[d], (void*)pge, (void*)ge);
 			gex_dump_contour(ge, clen);
 		} else {
 			/* when we chop off a piece of fragment, we leave the remaining
@@ -657,7 +657,7 @@ frag_subtract(
 			 * of the line fragment under consideration
 			 */
 			fprintf(stderr, "    cut %s frag at %p from len=%d to len=%d (end %p)\n", 
-				gxf_name[d], pge, pf->len[d], len+1, ge);
+				gxf_name[d], (void*)pge, pf->len[d], len+1, (void*)ge);
 			j = pf->len[d] - len - 1; /* how many gentries are chopped off */
 			pf->len[d] = len + 1;
 			i = slen - 1;
@@ -671,11 +671,11 @@ frag_subtract(
 				 */
 
 				fprintf(stderr, "    end of %s frag len=%d %p-", 
-					gxf_name[d], j+1, pge->bkwd);
+					gxf_name[d], j+1, (void*)pge->bkwd);
 				X_FRAG(pge->bkwd)->len[d] = j+1; /* the overlapping */
 				for(i = 1; j > 0; j--, i++, pge = pge->frwd)
 					X_FRAG(pge)->lenback[d] = i;
-				fprintf(stderr, "%p\n", pge->bkwd);
+				fprintf(stderr, "%p\n", (void*)pge->bkwd);
 				gex_dump_contour(ge, clen);
 			}
 		}
@@ -697,7 +697,7 @@ frag_subtract(
 			j = pf->len[d]; 
 			if(j != 0) {
 				fprintf(stderr, "    removed %s frag at %p len=%d\n", 
-					gxf_name[d], pge, j);
+					gxf_name[d], (void*)pge, j);
 				gex_dump_contour(ge, clen);
 				pf->len[d] = 0;
 				j--;
@@ -707,11 +707,11 @@ frag_subtract(
 	/* pge points at the last gentry of the line fragment */
 	if(j > 1) { /* we have the tail of a fragment left */
 		fprintf(stderr, "    end of %s frag len=%d %p-", 
-			gxf_name[d], j, pge);
+			gxf_name[d], j, (void*)pge);
 		X_FRAG(pge)->len[d] = j; /* the overlapping */
 		for(i = 0; j > 0; j--, i++, pge = pge->frwd)
 			X_FRAG(pge)->lenback[d] = i;
-		fprintf(stderr, "%p\n", pge->bkwd);
+		fprintf(stderr, "%p\n", (void*)pge->bkwd);
 		gex_dump_contour(ge, clen);
 	} else if(j == 1) {
 		X_FRAG(pge)->lenback[d] = 0;
@@ -1098,7 +1098,7 @@ bmp_outline(
 						if( isign(ge->frwd->ipoints[i][2] - ge->ipoints[i][2])
 						* isign(ge->bkwd->bkwd->ipoints[i][2] - ge->bkwd->ipoints[i][2]) == 1) {
 							X_FRAG(ge)->flags |= GEXFF_EXTR;
-							fprintf(stderr, "  %s extremum at %p\n", (i?"vert":"hor"), ge);
+							fprintf(stderr, "  %s extremum at %p\n", (i?"vert":"hor"), (void*)ge);
 						}
 						if(abs(ge->ipoints[i][2] - ge->bkwd->ipoints[i][2]) > 1)
 							X_FRAG(ge)->flags |= GEXFF_LONG;
@@ -1184,7 +1184,7 @@ bmp_outline(
 		dy = (ge)->iy3 - (ge)->bkwd->bkwd->iy3; \
 		if(0 && msg) { \
 			fprintf(stderr, "  %p: dx=%d dy=%d dx0=%d dy0=%d ", \
-				(ge), dx, dy, lastdx, lastdy); \
+				((void*)ge), dx, dy, lastdx, lastdy); \
 		} \
 		k1 = X_FRAG(ge)->flags; \
 		k2 = X_FRAG((ge)->bkwd)->flags; \
@@ -1213,7 +1213,7 @@ bmp_outline(
 		} \
 		if(0 && msg) { \
 			fprintf(stderr, "k1=%d k2=%d pge=%p count=%d %s good=%d rev=%d\n", \
-				k1, k2, pge, count, gxf_name[d], good, reversal); \
+				k1, k2, (void*)pge, count, gxf_name[d], good, reversal); \
 		} \
 	} while(0)
 
@@ -1765,7 +1765,7 @@ bmp_outline(
 						if(f->len[d] == 3) {
 							pge = age[(f->aidx + 2)%clen]; /* last gentry of this frag */
 							if(f->lenback[d] == 0 && X_FRAG(pge)->len[d] == 0) {
-								fprintf(stderr, "    discarded small %s at %p-%p\n", gxf_name[d], ge, pge);
+								fprintf(stderr, "    discarded small %s at %p-%p\n", gxf_name[d], (void*)ge, (void*)pge);
 								f->len[d] = 0;
 								X_FRAG(ge->frwd)->lenback[d] = 0;
 								X_FRAG(ge->frwd->frwd)->lenback[d] = 0;
@@ -1813,7 +1813,7 @@ bmp_outline(
 						continue;
 					}
 
-					fprintf(stderr, "   line at %p len=%d\n", ge, f->len[GEXFI_EXACTLINE]);
+					fprintf(stderr, "   line at %p len=%d\n", (void*)ge, f->len[GEXFI_EXACTLINE]);
 					for(d = GEXFI_CONVEX; d<= GEXFI_CONCAVE; d++) {
 						frag_subtract(g, age, clen, ge, len, d);
 					}
@@ -1855,7 +1855,7 @@ bmp_outline(
 								if(f->len[d] == len || f->len[d] == len+1) {
 
 									fprintf(stderr, "    removed %s frag at %p len=%d linelen=%d\n", 
-										gxf_name[d], ge, f->len[d], len);
+										gxf_name[d], (void*)ge, f->len[d], len);
 									pge = ge->frwd;
 									for(i = f->len[d]; i > 1; i--, pge = pge->frwd)
 										X_FRAG(pge)->lenback[d] = 0;
@@ -1864,7 +1864,7 @@ bmp_outline(
 									done = 1;
 								} else if(pf->len[d] == len+1 || pf->len[d] == len+sharpness) {
 									fprintf(stderr, "    removed %s frag at %p len=%d next linelen=%d\n", 
-										gxf_name[d], ge->bkwd, pf->len[d], len);
+										gxf_name[d], (void*)ge->bkwd, pf->len[d], len);
 									pge = ge;
 									for(i = pf->len[d]; i > 1; i--, pge = pge->frwd)
 										X_FRAG(pge)->lenback[d] = 0;
@@ -1935,7 +1935,7 @@ bmp_outline(
 					|| f->len[GEXFI_CONCAVE] >= len) {
 				line_completely_covered:
 						fprintf(stderr, "    removed covered Line frag at %p len=%d\n", 
-							ge, len);
+							(void*)ge, len);
 						f->len[GEXFI_LINE] = 0;
 						for(pge = ge->frwd; len > 1; len--, pge = pge->frwd)
 							X_FRAG(pge)->lenback[GEXFI_LINE] = 0;
@@ -1986,7 +1986,7 @@ bmp_outline(
 					}
 					if(k1 != 0 || k2 != 0) {
 						fprintf(stderr, "    cut Line frag at %p by (%d,%d) to len=%d\n", 
-							ge, k1, k2, len);
+							(void*)ge, k1, k2, len);
 						gex_dump_contour(ge, clen);
 
 						goto reconsider_line; /* the line may have to be cut again */
@@ -1997,7 +1997,7 @@ bmp_outline(
 						for(d = GEXFI_CONVEX; d<= GEXFI_CONCAVE; d++) {
 							if(X_FRAG(pge)->len[d]) {
 								fprintf(stderr, "    removed %s frag at %p len=%d covered by line %d\n", 
-									gxf_name[d], pge, X_FRAG(pge)->len[d], len);
+									gxf_name[d], (void*)pge, X_FRAG(pge)->len[d], len);
 								good = 0;
 							}
 							X_FRAG(pge)->len[d] = 0;
@@ -2036,7 +2036,7 @@ bmp_outline(
 							curve_completely_covered:
 								/* remove the convex frag */
 								fprintf(stderr, "    removed %s frag at %p len=%d covered by %s\n", 
-									gxf_name[d], ge, len, gxf_name[dx]);
+									gxf_name[d], (void*)ge, len, gxf_name[dx]);
 								f->len[d] = 0;
 								for(pge = ge->frwd, j = 1; j < len; j++, pge = pge->frwd)
 									X_FRAG(pge)->lenback[d] = 0;
@@ -2047,7 +2047,7 @@ bmp_outline(
 							} else {
 								/* remove the concave frag */
 								fprintf(stderr, "    removed %s frag at %p len=%d covered by %s\n", 
-									gxf_name[dx], ge, i, gxf_name[d]);
+									gxf_name[dx], (void*)ge, i, gxf_name[d]);
 								f->len[dx] = 0;
 								for(pge = ge->frwd, j = 1; j < i; j++, pge = pge->frwd)
 									X_FRAG(pge)->lenback[dx] = 0;
@@ -2069,7 +2069,7 @@ bmp_outline(
 							/* i >= 2 by definition */
 							if(i >= k2-1) { /* covers the other frag - maybe with 1 gentry showing */
 								fprintf(stderr, "    removed %s frag at %p len=%d covered by %s\n", 
-									gxf_name[dx], pge, k2, gxf_name[d]);
+									gxf_name[dx], (void*)pge, k2, gxf_name[d]);
 								X_FRAG(pge)->len[dx] = 0;
 								for(pge = pge->frwd, j = 1; j < k2; j++, pge = pge->frwd)
 									X_FRAG(pge)->lenback[dx] = 0;
@@ -2077,7 +2077,7 @@ bmp_outline(
 									/* our frag will be removed as well, prepare a line to replace it */
 									gels = ge;
 									gele = age[(f->aidx + i - 1)%clen];
-									fprintf(stderr, "    new Line frag at %p-%p len=%d\n", gels, gele, i);
+									fprintf(stderr, "    new Line frag at %p-%p len=%d\n", (void*)gels, (void*)gele, i);
 									X_FRAG(gels)->len[GEXFI_LINE] = i;
 									for(gei = gels->frwd, j = 1; j < i; gei = gei->frwd, j++)
 										X_FRAG(gei)->lenback[GEXFI_LINE] = j;
@@ -2120,7 +2120,7 @@ bmp_outline(
 							if(j != k2) {
 								X_FRAG(pge)->len[dx] = j;
 								fprintf(stderr, "    cut %s frag at %p len=%d to %p len=%d end overlap with %s\n", 
-									gxf_name[dx], pge, k2, gels, j, gxf_name[d]);
+									gxf_name[dx], (void*)pge, k2, (void*)gels, j, gxf_name[d]);
 								for(gei = gels->frwd; j < k2; gei = gei->frwd, j++)
 									X_FRAG(gei)->lenback[dx] = 0;
 							}
@@ -2128,20 +2128,20 @@ bmp_outline(
 							if(gele != ge) {
 								sge = gele;
 								f->len[d] = 0;
-								fprintf(stderr, "    cut %s frag at %p len=%d ", gxf_name[d], ge, len);
+								fprintf(stderr, "    cut %s frag at %p len=%d ", gxf_name[d], (void*)ge, len);
 								len--;
 								for(gei = ge->frwd; gei != gele; gei = gei->frwd, len--)
 									X_FRAG(gei)->lenback[d] = 0;
 								X_FRAG(gele)->len[d] = len;
 								X_FRAG(gele)->lenback[d] = 0;
 								fprintf(stderr, "to %p len=%d start overlap with %s\n", 
-									sge, len, gxf_name[dx]);
+									(void*)sge, len, gxf_name[dx]);
 								for(gei = gei->frwd, j = 1; j < len; gei = gei->frwd, j++)
 									X_FRAG(gei)->lenback[d] = j;
 
 							}
 							if(i > 1) {
-								fprintf(stderr, "    new Line frag at %p-%p len=%d\n", gels, gele, i);
+								fprintf(stderr, "    new Line frag at %p-%p len=%d\n", (void*)gels, (void*)gele, i);
 								X_FRAG(gels)->len[GEXFI_LINE] = i;
 								for(gei = gels->frwd, j = 1; j < i; gei = gei->frwd, j++)
 									X_FRAG(gei)->lenback[GEXFI_LINE] = j;
@@ -2171,7 +2171,7 @@ bmp_outline(
 							if(dx >= 0) {
 								fprintf(stderr, "**** Internal error in vectorization\n");
 								fprintf(stderr, "CONFLICT in %s at %p between %s and %s\n",
-									g->name, ge, gxf_name[dx], gxf_name[d]);
+									g->name, (void*)ge, gxf_name[dx], gxf_name[d]);
 								dumppaths(g, cge->next, cge->next->bkwd);
 								gex_dump_contour(ge, clen);
 								exit(1);
@@ -2182,7 +2182,7 @@ bmp_outline(
 							if(dy >= 0) {
 								fprintf(stderr, "**** Internal error in vectorization\n");
 								fprintf(stderr, "CONFLICT in %s at %p between %s and %s\n",
-									g->name, ge, gxf_name[dy], gxf_name[d]);
+									g->name, (void*)ge, gxf_name[dy], gxf_name[d]);
 								dumppaths(g, cge->next, cge->next->bkwd);
 								gex_dump_contour(ge, clen);
 								exit(1);
@@ -2360,7 +2360,7 @@ bmp_outline(
 						}
 
 						fprintf(stderr, " %s frag %p%s nsub=%d\n", gxf_name[f->ixstart],
-							ge, (f->flags&GEXFF_CIRC)?" circular":"", nsub);
+							(void*)ge, (f->flags&GEXFF_CIRC)?" circular":"", nsub);
 
 						/* find the symmetry between the subfragments */
 						for(gef = firstge, count=0; count < nsub; gef = ff->nextsub, count++) {
@@ -2480,7 +2480,7 @@ bmp_outline(
 							/* debugging stuff */
 							ff = X_FRAG(gef);
 							fprintf(stderr, "   %p-%p bbox[%g,%g] sym=%p %s len=%d xlen=%d\n",
-								gef, ff->nextsub, ff->bbox[0], ff->bbox[1], ff->symge, 
+								(void*)gef, (void*)ff->nextsub, ff->bbox[0], ff->bbox[1], (void*)ff->symge, 
 								(ff->flags & GEXFF_SYMNEXT) ? "symnext" : "",
 								ff->sublen, ff->symxlen);
 

--- a/src/ttf2pt1/pt1.c
+++ b/src/ttf2pt1/pt1.c
@@ -298,13 +298,13 @@ assertpath(
 		if( (ge->flags & GEF_FLOAT) ^ isfloat ) {
 			fprintf(stderr, "**! assertpath: called from %s line %d (%s) ****\n", file, line, name);
 			fprintf(stderr, "float flag changes from %s to %s at 0x%p (type %c, prev type %c)\n",
-				(isfloat ? "TRUE" : "FALSE"), (isfloat ? "FALSE" : "TRUE"), ge, ge->type, pe->type);
+				(isfloat ? "TRUE" : "FALSE"), (isfloat ? "FALSE" : "TRUE"), (void*)ge, ge->type, pe->type);
 			abort();
 		}
 		if (pe != ge->prev) {
 			fprintf(stderr, "**! assertpath: called from %s line %d (%s) ****\n", file, line, name);
 			fprintf(stderr, "unidirectional chain 0x%p -next-> 0x%p -prev-> 0x%p \n",
-				pe, ge, ge->prev);
+				(void*)pe, (void*)ge, (void*)ge->prev);
 			abort();
 		}
 
@@ -314,7 +314,7 @@ assertpath(
 		case GE_PATH:
 			if (ge->prev == 0) {
 				fprintf(stderr, "**! assertpath: called from %s line %d (%s) ****\n", file, line, name);
-				fprintf(stderr, "empty path at 0x%p \n", ge);
+				fprintf(stderr, "empty path at 0x%p \n", (void*)ge);
 				abort();
 			}
 			break;
@@ -323,7 +323,7 @@ assertpath(
 			if(ge->frwd->bkwd != ge) {
 				fprintf(stderr, "**! assertpath: called from %s line %d (%s) ****\n", file, line, name);
 				fprintf(stderr, "unidirectional chain 0x%p -frwd-> 0x%p -bkwd-> 0x%p \n",
-					ge, ge->frwd, ge->frwd->bkwd);
+					(void*)ge, (void*)ge->frwd, (void*)ge->frwd->bkwd);
 				abort();
 			}
 			if(ge->prev->type == GE_MOVE) {
@@ -331,7 +331,7 @@ assertpath(
 				if(ge->bkwd->next->type != GE_PATH) {
 					fprintf(stderr, "**! assertpath: called from %s line %d (%s) ****\n", file, line, name);
 					fprintf(stderr, "broken first backlink 0x%p -bkwd-> 0x%p -next-> 0x%p \n",
-						ge, ge->bkwd, ge->bkwd->next);
+						(void*)ge, (void*)ge->bkwd, (void*)ge->bkwd->next);
 					abort();
 				}
 			}
@@ -339,7 +339,7 @@ assertpath(
 				if(ge->frwd != first) {
 					fprintf(stderr, "**! assertpath: called from %s line %d (%s) ****\n", file, line, name);
 					fprintf(stderr, "broken loop 0x%p -...-> 0x%p -frwd-> 0x%p \n",
-						first, ge, ge->frwd);
+						(void*)first, (void*)ge, (void*)ge->frwd);
 					abort();
 				}
 			}
@@ -760,7 +760,7 @@ fixcvends(
 		return;
 
 	if(ge->flags & GEF_FLOAT) {
-		fprintf(stderr, "**! fixcvends(0x%p) on floating entry, ABORT\n", ge);
+		fprintf(stderr, "**! fixcvends(0x%p) on floating entry, ABORT\n", (void*)ge);
 		abort(); /* dump core */
 	}
 
@@ -910,7 +910,7 @@ fixcvdir(
 	int             fdir, rdir;
 
 	if(ge->flags & GEF_FLOAT) {
-		fprintf(stderr, "**! fixcvdir(0x%p) on floating entry, ABORT\n", ge);
+		fprintf(stderr, "**! fixcvdir(0x%p) on floating entry, ABORT\n", (void*)ge);
 		abort(); /* dump core */
 	}
 
@@ -1040,7 +1040,7 @@ fgetcvdir(
 	int             dir = 0;
 
 	if( !(ge->flags & GEF_FLOAT) ) {
-		fprintf(stderr, "**! fgetcvdir(0x%p) on int entry, ABORT\n", ge);
+		fprintf(stderr, "**! fgetcvdir(0x%p) on int entry, ABORT\n", (void*)ge);
 		abort(); /* dump core */
 	}
 
@@ -1102,7 +1102,7 @@ igetcvdir(
 	int             dir = 0;
 
 	if(ge->flags & GEF_FLOAT) {
-		fprintf(stderr, "**! igetcvdir(0x%p) on floating entry, ABORT\n", ge);
+		fprintf(stderr, "**! igetcvdir(0x%p) on floating entry, ABORT\n", (void*)ge);
 		abort(); /* dump core */
 	}
 
@@ -1200,7 +1200,7 @@ dumppaths(
 	for(ge = g->entries; ge != 0; ge = ge->next) {
 		if(ge == start)
 			mark = '*';
-		fprintf(stderr, " %c %8p", mark, ge);
+		fprintf(stderr, " %c %8p", mark, (void*)ge);
 		switch(ge->type) {
 		case GE_MOVE:
 		case GE_LINE:
@@ -1333,7 +1333,7 @@ fcheckcv(
 )
 {
 	if( !(ge->flags & GEF_FLOAT) ) {
-		fprintf(stderr, "**! fcheckcv(0x%p) on int entry, ABORT\n", ge);
+		fprintf(stderr, "**! fcheckcv(0x%p) on int entry, ABORT\n", (void*)ge);
 		abort(); /* dump core */
 	}
 
@@ -1361,7 +1361,7 @@ icheckcv(
 	int             xdep, ydep;
 
 	if(ge->flags & GEF_FLOAT) {
-		fprintf(stderr, "**! icheckcv(0x%p) on floating entry, ABORT\n", ge);
+		fprintf(stderr, "**! icheckcv(0x%p) on floating entry, ABORT\n", (void*)ge);
 		abort(); /* dump core */
 	}
 
@@ -2918,7 +2918,7 @@ findstemat(
 			if(ISDBG(SUBSTEMS)) {
 				fprintf(stderr, 
 					"dbg: possible self-intersection at v=%d o=%d exp_ge=0x%p ge=0x%p\n",
-					value, origin, ge, sp[si].ge);
+					value, origin, (void*)ge, (void*)sp[si].ge);
 			}
 			continue;
 		}
@@ -3909,7 +3909,7 @@ fstraighten(
 			if(ge != pge) { 
 				if( pge->type == GE_LINE && pge->fpoints[i][2] == pge->prev->fpoints[i][2]
 				&& abs(pge->fpoints[o][2] != pge->prev->fpoints[o][2]) ) {
-					if(ISDBG(STRAIGHTEN)) fprintf(stderr,"** straighten join with previous 0x%p 0x%p\n", pge, ge);
+					if(ISDBG(STRAIGHTEN)) fprintf(stderr,"** straighten join with previous 0x%p 0x%p\n", (void*)pge, (void*)ge);
 					/* join the previous line with current */
 					pge->fx3 = ge->fx3;
 					pge->fy3 = ge->fy3;
@@ -3924,7 +3924,7 @@ fstraighten(
 			if(ge != nge) { 
 				if (nge->type == GE_LINE && nge->fpoints[i][2] == ge->fpoints[i][2]
 				&& abs(nge->fpoints[o][2] != ge->fpoints[o][2]) ) {
-					if(ISDBG(STRAIGHTEN)) fprintf(stderr,"** straighten join with next 0x%p 0x%p\n", ge, nge);
+					if(ISDBG(STRAIGHTEN)) fprintf(stderr,"** straighten join with next 0x%p 0x%p\n", (void*)ge, (void*)nge);
 					/* join the next line with current */
 					ge->fx3 = nge->fx3;
 					ge->fy3 = nge->fy3;
@@ -4056,7 +4056,7 @@ ffixquadrants(
 		np = 0; /* no split points yet */
 		if(ISDBG(QUAD)) {
 			fprintf(stderr, "%s: trying 0x%p (%g %g) (%g %g) (%g %g) (%g %g)\n  ", g->name,
-				ge,  ge->prev->fx3, ge->prev->fy3, ge->fx1, ge->fy1, ge->fx2, ge->fy2,
+				(void*)ge,  ge->prev->fx3, ge->prev->fy3, ge->fx1, ge->fy1, ge->fx2, ge->fy2,
 				ge->fx3, ge->fy3);
 		}
 		for(i=0; i<2; i++) { /* first for x then for y */
@@ -4077,7 +4077,7 @@ ffixquadrants(
 
 			if(ISDBG(QUAD))
 				fprintf(stderr, "%s: 0x%p: %d pts(%c): ", 
-					g->name, ge, np-oldnp, i? 'y':'x');
+					g->name, (void*)ge, np-oldnp, i? 'y':'x');
 
 			/* remove points that are too close to the ends 
 			 * because hor/vert ends are permitted, also
@@ -4128,7 +4128,7 @@ ffixquadrants(
 
 		if(ISDBG(QUAD)) {
 			fprintf(stderr, "%s: splitting 0x%p (%g %g) (%g %g) (%g %g) (%g %g) at %d points\n  ", g->name,
-				ge,  ge->prev->fx3, ge->prev->fy3, ge->fx1, ge->fy1, ge->fx2, ge->fy2,
+				(void*)ge,  ge->prev->fx3, ge->prev->fy3, ge->fx1, ge->fy1, ge->fx2, ge->fy2,
 				ge->fx3, ge->fy3, np);
 			for(i=0; i<np; i++)
 				fprintf(stderr, "%g(%c) ", sp[i], dir[i] ? 'y':'x');
@@ -4149,7 +4149,7 @@ ffixquadrants(
 			k1 = sp[j];
 			k2 = 1 - k1;
 
-			if(ISDBG(QUAD)) fprintf(stderr, "   0x%p %g/%g\n", ge, k1, k2);
+			if(ISDBG(QUAD)) fprintf(stderr, "   0x%p %g/%g\n", (void*)ge, k1, k2);
 
 			nge = newgentry(GEF_FLOAT);
 			(*nge) = (*ge);
@@ -4690,7 +4690,7 @@ fdelsmall(
 
 		if(ISDBG(FCONCISE))  {
 			fprintf(stderr, "glyph %s has very small fragments(%p..%p..%p)\n", 
-			g->name, pge, ge, nge);
+			g->name, (void*)pge, (void*)ge, (void*)nge);
 			dumppaths(g, pge, nge);
 		}
 
@@ -5864,7 +5864,7 @@ fconcisecontour(
 		return; /* probably a degenerate contour */
 
 	if(ISDBG(FCONCISE))
-		fprintf(stderr, "processing contour 0x%p of glyph %s\n", startge, g->name);
+		fprintf(stderr, "processing contour 0x%p of glyph %s\n", (void*)startge, g->name);
 
 	maxdots = MAXDOTS;
 	dots = (struct dot_dist *)malloc(sizeof(*dots)*maxdots);
@@ -5881,7 +5881,7 @@ fconcisecontour(
 		if((gex->flags & GEXF_JMASK) > ((joinmask<<1)-1)) {
 			if(ISDBG(FCONCISE))
 				fprintf(stderr, "found higher flag (%x>%x) at 0x%p\n", 
-					gex->flags & GEXF_JMASK, ((joinmask<<1)-1), ge);
+					gex->flags & GEXF_JMASK, ((joinmask<<1)-1), (void*)ge);
 			joinmask <<= 1;
 			startge = ge; /* have to redo the pass */
 			continue;
@@ -5903,7 +5903,7 @@ fconcisecontour(
 		pgex = X_CON(pge->bkwd);
 
 		if(ISDBG(FCONCISE))
-			fprintf(stderr, "ge %p prev -> 0x%p ", ge, pge);
+			fprintf(stderr, "ge %p prev -> 0x%p ", (void*)ge, (void*)pge);
 
 		while(pgex->flags & GEXF_JCVMASK) {
 			if( !(pgex->flags & ((GEXF_JCVMASK^GEXF_JID)|GEXF_JID2)) )
@@ -5920,7 +5920,7 @@ fconcisecontour(
 					/* the previos entry is definitely a better match */
 					if(pge == ge) {
 						if(ISDBG(FCONCISE))
-							fprintf(stderr, "\nprev is a better match at %p\n", pge);
+							fprintf(stderr, "\nprev is a better match at %p\n", (void*)pge);
 						startge = ge;
 						goto next;
 					} else
@@ -5933,7 +5933,7 @@ fconcisecontour(
 			pge = pge->bkwd;
 			pgex = X_CON(pge->bkwd);
 			if(ISDBG(FCONCISE))
-				fprintf(stderr, "0x%p ", pge);
+				fprintf(stderr, "0x%p ", (void*)pge);
 		}
 
 		/* collect as many entries for joining as possible */
@@ -5942,7 +5942,7 @@ fconcisecontour(
 		nngex = X_CON(nge->frwd); 
 
 		if(ISDBG(FCONCISE))
-			fprintf(stderr, ": 0x%p\nnext -> 0x%p ", pge, nge);
+			fprintf(stderr, ": 0x%p\nnext -> 0x%p ", (void*)pge, (void*)nge);
 
 		while(ngex->flags & GEXF_JCVMASK) {
 			if( !(ngex->flags & ((GEXF_JCVMASK^GEXF_JID)|GEXF_JID1)) )
@@ -5959,7 +5959,7 @@ fconcisecontour(
 					if(nge == ge->frwd) {
 						if(ISDBG(FCONCISE))
 							fprintf(stderr, "\nnext %x is a better match than %x at %p (jmask %x)\n", 
-								ngex->flags & GEXF_JCVMASK, gex->flags & GEXF_JCVMASK, nge, joinmask);
+								ngex->flags & GEXF_JCVMASK, gex->flags & GEXF_JCVMASK, (void*)nge, joinmask);
 						goto next;
 					} else
 						nge = nge->bkwd;
@@ -5972,11 +5972,11 @@ fconcisecontour(
 			ngex = nngex; 
 			nngex = X_CON(nge->frwd);
 			if(ISDBG(FCONCISE))
-				fprintf(stderr, "0x%p ", nge);
+				fprintf(stderr, "0x%p ", (void*)nge);
 		}
 
 		if(ISDBG(FCONCISE))
-			fprintf(stderr, ": 0x%p\n", nge);
+			fprintf(stderr, ": 0x%p\n", (void*)nge);
 
 		/* XXX add splitting of last entries if neccessary */
 
@@ -6078,7 +6078,7 @@ fconcisecontour(
 					ggoodcv++; ggoodcvdots += ndots;
 
 					if(ISDBG(FCONCISE)) {
-						fprintf(stderr, "in %s joined %p-%p to ", g->name, pge, nge);
+						fprintf(stderr, "in %s joined %p-%p to ", g->name, (void*)pge, (void*)nge);
 						for(i=0; i<4; i++) {
 							fprintf(stderr, " (%g, %g)", apcv[i][X], apcv[i][Y]);
 						}
@@ -6145,7 +6145,7 @@ fconcisecontour(
 				nge = nge->bkwd;
 			}
 			if(ISDBG(FCONCISE))
-				fprintf(stderr, "next try: %p to %p\n", pge, nge);
+				fprintf(stderr, "next try: %p to %p\n", (void*)pge, (void*)nge);
 		}
 
 next:
@@ -6173,17 +6173,17 @@ next:
 		nge = ge->frwd;
 
 		if(ISDBG(FCONCISE))
-			fprintf(stderr, "joining LINE from %p-%p\n", ge, nge);
+			fprintf(stderr, "joining LINE from %p-%p\n", (void*)ge, (void*)nge);
 
 		while(pge!=nge) {
 			pgex = X_CON(pge);
 			ngex = X_CON(nge); 
 			if(ISDBG(FCONCISE))
-				fprintf(stderr, "(p=%p/%x n=0x%p/%x) ", pge, pgex->flags & GEXF_JLINE, 
-					nge, ngex->flags & GEXF_JLINE);
+				fprintf(stderr, "(p=%p/%x n=0x%p/%x) ", (void*)pge, pgex->flags & GEXF_JLINE, 
+					(void*)nge, ngex->flags & GEXF_JLINE);
 			if( !((pgex->flags | ngex->flags) & GEXF_JLINE) ) {
 				if(ISDBG(FCONCISE))
-					fprintf(stderr, "(end p=%p n=%p) ", pge, nge);
+					fprintf(stderr, "(end p=%p n=%p) ", (void*)pge, (void*)nge);
 				break;
 			}
 
@@ -6209,19 +6209,19 @@ next:
 				}
 				if(i<ndots) { /* failed to join */
 					if(ISDBG(FCONCISE))
-						fprintf(stderr, "failed to join prev %p ", pge);
+						fprintf(stderr, "failed to join prev %p ", (void*)pge);
 					ndots--;
 					pgex->flags &= ~GEXF_JLINE;
 				} else {
 					pge = pge->bkwd;
 					if(pge == nge) {
 						if(ISDBG(FCONCISE))
-							fprintf(stderr, "intersected at prev %p ", pge);
+							fprintf(stderr, "intersected at prev %p ", (void*)pge);
 						break; /* oops, tried to self-intersect */
 					}
 				}
 			} else if(ISDBG(FCONCISE))
-				fprintf(stderr, "(p=%p) ", pge);
+				fprintf(stderr, "(p=%p) ", (void*)pge);
 
 			if( ngex->flags & GEXF_JLINE ) {
 				for(i=0; i<2; i++) {
@@ -6237,14 +6237,14 @@ next:
 				}
 				if(i<ndots) { /* failed to join */
 					if(ISDBG(FCONCISE))
-						fprintf(stderr, "failed to join next %p ", nge->frwd);
+						fprintf(stderr, "failed to join next %p ", (void*)nge->frwd);
 					ndots--;
 					ngex->flags &= ~GEXF_JLINE;
 				} else {
 					nge = nge->frwd;
 				}
 			} else if(ISDBG(FCONCISE))
-				fprintf(stderr, "(n=%p) ", nge);
+				fprintf(stderr, "(n=%p) ", (void*)nge);
 		}
 
 		pge = pge->frwd;  /* now the limits are pge...nge inclusive */
@@ -6252,7 +6252,7 @@ next:
 			break;
 
 		if(ISDBG(FCONCISE)) {
-			fprintf(stderr, "\nin %s joined LINE %p-%p from\n", g->name, pge, nge);
+			fprintf(stderr, "\nin %s joined LINE %p-%p from\n", g->name, (void*)pge, (void*)nge);
 			dumppaths(g, pge, nge);
 		}
 		pge->type = GE_LINE;
@@ -7068,7 +7068,7 @@ reversepathsfromto(
 	for (ge = from; ge != 0 && ge != to; ge = ge->next) {
 		if(ge->type == GE_LINE || ge->type == GE_CURVE) {
 			if (ISDBG(REVERSAL))
-				fprintf(stderr, "reverse path 0x%p <- 0x%p, 0x%p\n", ge, ge->prev, ge->bkwd);
+				fprintf(stderr, "reverse path 0x%p <- 0x%p, 0x%p\n", (void*)ge, (void*)ge->prev, (void*)ge->bkwd);
 
 			/* cut out the path itself */
 			pge = ge->prev; /* GE_MOVE */

--- a/src/ttf2pt1/pt1.c
+++ b/src/ttf2pt1/pt1.c
@@ -303,7 +303,7 @@ assertpath(
 		}
 		if (pe != ge->prev) {
 			fprintf(stderr, "**! assertpath: called from %s line %d (%s) ****\n", file, line, name);
-			fprintf(stderr, "unidirectional chain 0x%x -next-> 0x%x -prev-> 0x%x \n",
+			fprintf(stderr, "unidirectional chain 0x%p -next-> 0x%p -prev-> 0x%p \n",
 				pe, ge, ge->prev);
 			abort();
 		}
@@ -314,7 +314,7 @@ assertpath(
 		case GE_PATH:
 			if (ge->prev == 0) {
 				fprintf(stderr, "**! assertpath: called from %s line %d (%s) ****\n", file, line, name);
-				fprintf(stderr, "empty path at 0x%x \n", ge);
+				fprintf(stderr, "empty path at 0x%p \n", ge);
 				abort();
 			}
 			break;
@@ -322,7 +322,7 @@ assertpath(
 		case GE_CURVE:
 			if(ge->frwd->bkwd != ge) {
 				fprintf(stderr, "**! assertpath: called from %s line %d (%s) ****\n", file, line, name);
-				fprintf(stderr, "unidirectional chain 0x%x -frwd-> 0x%x -bkwd-> 0x%x \n",
+				fprintf(stderr, "unidirectional chain 0x%p -frwd-> 0x%p -bkwd-> 0x%p \n",
 					ge, ge->frwd, ge->frwd->bkwd);
 				abort();
 			}
@@ -330,7 +330,7 @@ assertpath(
 				first = ge;
 				if(ge->bkwd->next->type != GE_PATH) {
 					fprintf(stderr, "**! assertpath: called from %s line %d (%s) ****\n", file, line, name);
-					fprintf(stderr, "broken first backlink 0x%x -bkwd-> 0x%x -next-> 0x%x \n",
+					fprintf(stderr, "broken first backlink 0x%p -bkwd-> 0x%p -next-> 0x%p \n",
 						ge, ge->bkwd, ge->bkwd->next);
 					abort();
 				}
@@ -338,7 +338,7 @@ assertpath(
 			if(ge->next->type == GE_PATH) {
 				if(ge->frwd != first) {
 					fprintf(stderr, "**! assertpath: called from %s line %d (%s) ****\n", file, line, name);
-					fprintf(stderr, "broken loop 0x%x -...-> 0x%x -frwd-> 0x%x \n",
+					fprintf(stderr, "broken loop 0x%p -...-> 0x%p -frwd-> 0x%p \n",
 						first, ge, ge->frwd);
 					abort();
 				}
@@ -760,7 +760,7 @@ fixcvends(
 		return;
 
 	if(ge->flags & GEF_FLOAT) {
-		fprintf(stderr, "**! fixcvends(0x%x) on floating entry, ABORT\n", ge);
+		fprintf(stderr, "**! fixcvends(0x%p) on floating entry, ABORT\n", ge);
 		abort(); /* dump core */
 	}
 
@@ -910,7 +910,7 @@ fixcvdir(
 	int             fdir, rdir;
 
 	if(ge->flags & GEF_FLOAT) {
-		fprintf(stderr, "**! fixcvdir(0x%x) on floating entry, ABORT\n", ge);
+		fprintf(stderr, "**! fixcvdir(0x%p) on floating entry, ABORT\n", ge);
 		abort(); /* dump core */
 	}
 
@@ -1040,7 +1040,7 @@ fgetcvdir(
 	int             dir = 0;
 
 	if( !(ge->flags & GEF_FLOAT) ) {
-		fprintf(stderr, "**! fgetcvdir(0x%x) on int entry, ABORT\n", ge);
+		fprintf(stderr, "**! fgetcvdir(0x%p) on int entry, ABORT\n", ge);
 		abort(); /* dump core */
 	}
 
@@ -1102,7 +1102,7 @@ igetcvdir(
 	int             dir = 0;
 
 	if(ge->flags & GEF_FLOAT) {
-		fprintf(stderr, "**! igetcvdir(0x%x) on floating entry, ABORT\n", ge);
+		fprintf(stderr, "**! igetcvdir(0x%p) on floating entry, ABORT\n", ge);
 		abort(); /* dump core */
 	}
 
@@ -1200,7 +1200,7 @@ dumppaths(
 	for(ge = g->entries; ge != 0; ge = ge->next) {
 		if(ge == start)
 			mark = '*';
-		fprintf(stderr, " %c %8x", mark, ge);
+		fprintf(stderr, " %c %8p", mark, ge);
 		switch(ge->type) {
 		case GE_MOVE:
 		case GE_LINE:
@@ -1333,7 +1333,7 @@ fcheckcv(
 )
 {
 	if( !(ge->flags & GEF_FLOAT) ) {
-		fprintf(stderr, "**! fcheckcv(0x%x) on int entry, ABORT\n", ge);
+		fprintf(stderr, "**! fcheckcv(0x%p) on int entry, ABORT\n", ge);
 		abort(); /* dump core */
 	}
 
@@ -1361,7 +1361,7 @@ icheckcv(
 	int             xdep, ydep;
 
 	if(ge->flags & GEF_FLOAT) {
-		fprintf(stderr, "**! icheckcv(0x%x) on floating entry, ABORT\n", ge);
+		fprintf(stderr, "**! icheckcv(0x%p) on floating entry, ABORT\n", ge);
 		abort(); /* dump core */
 	}
 
@@ -2917,7 +2917,7 @@ findstemat(
 		if(sp[si].ge != ge) {
 			if(ISDBG(SUBSTEMS)) {
 				fprintf(stderr, 
-					"dbg: possible self-intersection at v=%d o=%d exp_ge=0x%x ge=0x%x\n",
+					"dbg: possible self-intersection at v=%d o=%d exp_ge=0x%p ge=0x%p\n",
 					value, origin, ge, sp[si].ge);
 			}
 			continue;
@@ -3909,7 +3909,7 @@ fstraighten(
 			if(ge != pge) { 
 				if( pge->type == GE_LINE && pge->fpoints[i][2] == pge->prev->fpoints[i][2]
 				&& abs(pge->fpoints[o][2] != pge->prev->fpoints[o][2]) ) {
-					if(ISDBG(STRAIGHTEN)) fprintf(stderr,"** straighten join with previous 0x%x 0x%x\n", pge, ge);
+					if(ISDBG(STRAIGHTEN)) fprintf(stderr,"** straighten join with previous 0x%p 0x%p\n", pge, ge);
 					/* join the previous line with current */
 					pge->fx3 = ge->fx3;
 					pge->fy3 = ge->fy3;
@@ -3924,7 +3924,7 @@ fstraighten(
 			if(ge != nge) { 
 				if (nge->type == GE_LINE && nge->fpoints[i][2] == ge->fpoints[i][2]
 				&& abs(nge->fpoints[o][2] != ge->fpoints[o][2]) ) {
-					if(ISDBG(STRAIGHTEN)) fprintf(stderr,"** straighten join with next 0x%x 0x%x\n", ge, nge);
+					if(ISDBG(STRAIGHTEN)) fprintf(stderr,"** straighten join with next 0x%p 0x%p\n", ge, nge);
 					/* join the next line with current */
 					ge->fx3 = nge->fx3;
 					ge->fy3 = nge->fy3;
@@ -4055,7 +4055,7 @@ ffixquadrants(
 	doagain:
 		np = 0; /* no split points yet */
 		if(ISDBG(QUAD)) {
-			fprintf(stderr, "%s: trying 0x%x (%g %g) (%g %g) (%g %g) (%g %g)\n  ", g->name,
+			fprintf(stderr, "%s: trying 0x%p (%g %g) (%g %g) (%g %g) (%g %g)\n  ", g->name,
 				ge,  ge->prev->fx3, ge->prev->fy3, ge->fx1, ge->fy1, ge->fx2, ge->fy2,
 				ge->fx3, ge->fy3);
 		}
@@ -4076,7 +4076,7 @@ ffixquadrants(
 				continue;
 
 			if(ISDBG(QUAD))
-				fprintf(stderr, "%s: 0x%x: %d pts(%c): ", 
+				fprintf(stderr, "%s: 0x%p: %d pts(%c): ", 
 					g->name, ge, np-oldnp, i? 'y':'x');
 
 			/* remove points that are too close to the ends 
@@ -4127,7 +4127,7 @@ ffixquadrants(
 			continue;
 
 		if(ISDBG(QUAD)) {
-			fprintf(stderr, "%s: splitting 0x%x (%g %g) (%g %g) (%g %g) (%g %g) at %d points\n  ", g->name,
+			fprintf(stderr, "%s: splitting 0x%p (%g %g) (%g %g) (%g %g) (%g %g) at %d points\n  ", g->name,
 				ge,  ge->prev->fx3, ge->prev->fy3, ge->fx1, ge->fy1, ge->fx2, ge->fy2,
 				ge->fx3, ge->fy3, np);
 			for(i=0; i<np; i++)
@@ -4149,7 +4149,7 @@ ffixquadrants(
 			k1 = sp[j];
 			k2 = 1 - k1;
 
-			if(ISDBG(QUAD)) fprintf(stderr, "   0x%x %g/%g\n", ge, k1, k2);
+			if(ISDBG(QUAD)) fprintf(stderr, "   0x%p %g/%g\n", ge, k1, k2);
 
 			nge = newgentry(GEF_FLOAT);
 			(*nge) = (*ge);
@@ -4689,7 +4689,7 @@ fdelsmall(
 		/* now we have a sequence of small fragments in pge...nge (inclusive) */
 
 		if(ISDBG(FCONCISE))  {
-			fprintf(stderr, "glyph %s has very small fragments(%x..%x..%x)\n", 
+			fprintf(stderr, "glyph %s has very small fragments(%p..%p..%p)\n", 
 			g->name, pge, ge, nge);
 			dumppaths(g, pge, nge);
 		}
@@ -5942,7 +5942,7 @@ fconcisecontour(
 		nngex = X_CON(nge->frwd); 
 
 		if(ISDBG(FCONCISE))
-			fprintf(stderr, ": 0x%x\nnext -> 0x%p ", pge, nge);
+			fprintf(stderr, ": 0x%p\nnext -> 0x%p ", pge, nge);
 
 		while(ngex->flags & GEXF_JCVMASK) {
 			if( !(ngex->flags & ((GEXF_JCVMASK^GEXF_JID)|GEXF_JID1)) )
@@ -5976,7 +5976,7 @@ fconcisecontour(
 		}
 
 		if(ISDBG(FCONCISE))
-			fprintf(stderr, ": 0x%x\n", nge);
+			fprintf(stderr, ": 0x%p\n", nge);
 
 		/* XXX add splitting of last entries if neccessary */
 
@@ -6179,7 +6179,7 @@ next:
 			pgex = X_CON(pge);
 			ngex = X_CON(nge); 
 			if(ISDBG(FCONCISE))
-				fprintf(stderr, "(p=%p/%x n=0x%x/%x) ", pge, pgex->flags & GEXF_JLINE, 
+				fprintf(stderr, "(p=%p/%x n=0x%p/%x) ", pge, pgex->flags & GEXF_JLINE, 
 					nge, ngex->flags & GEXF_JLINE);
 			if( !((pgex->flags | ngex->flags) & GEXF_JLINE) ) {
 				if(ISDBG(FCONCISE))
@@ -7068,7 +7068,7 @@ reversepathsfromto(
 	for (ge = from; ge != 0 && ge != to; ge = ge->next) {
 		if(ge->type == GE_LINE || ge->type == GE_CURVE) {
 			if (ISDBG(REVERSAL))
-				fprintf(stderr, "reverse path 0x%x <- 0x%x, 0x%x\n", ge, ge->prev, ge->bkwd);
+				fprintf(stderr, "reverse path 0x%p <- 0x%p, 0x%p\n", ge, ge->prev, ge->bkwd);
 
 			/* cut out the path itself */
 			pge = ge->prev; /* GE_MOVE */
@@ -7217,7 +7217,7 @@ print_kerning(
 		return;
 
 	fprintf(afm_file, "StartKernData\n");
-	fprintf(afm_file, "StartKernPairs %hd\n", kerning_pairs);
+	fprintf(afm_file, "StartKernPairs %d\n", kerning_pairs);
 
 	for(i=0; i<numglyphs; i++)  {
 		g = &glyph_list[i];

--- a/src/ttf2pt1/t1asm.c
+++ b/src/ttf2pt1/t1asm.c
@@ -371,7 +371,7 @@ static void charstring_byte(int v)
   byte b = (byte) (v & 0xff);
 
   if (charstring_bp - charstring_buf > sizeof(charstring_buf)) {
-    fprintf(stderr, "error: charstring_buf full (%d bytes)\n",
+    fprintf(stderr, "error: charstring_buf full (%zu bytes)\n",
             sizeof(charstring_buf));
     exit(1);
   }
@@ -385,7 +385,7 @@ static void charstring_end()
 {
   byte *bp;
 
-  sprintf(line, "%d ", charstring_bp - charstring_buf);
+  sprintf(line, "%ld ", charstring_bp - charstring_buf);
   eexec_string(line);
   sprintf(line, "%s ", cs_start);
   eexec_string(line);

--- a/src/ttf2pt1/ttf2pt1.c
+++ b/src/ttf2pt1/ttf2pt1.c
@@ -475,7 +475,7 @@ static struct uni_language uni_lang[]= {
 		{ "hu_","pl_","cz_","si_","sk_" },
 		'A'
 	},
-	{ /* by Rièardas Èepas <rch@WriteMe.Com> */
+	{ /* by Riï¿½ardas ï¿½epas <rch@WriteMe.Com> */
 		{ unicode_latin4 }, 
 		0, /* no name-based mapping */
 		"latin4",
@@ -644,7 +644,7 @@ unicode_init_user(
 			continue;
 		}
 
-		if(sscanf(buffer, "id %d %d", pid, eid)==2) {
+		if(sscanf(buffer, "id %d %d", &pid, &eid)==2) {
 			if( !overid /* only if the user has not overriden */
 			&& (enabled || !sawplane) ) { 
 				force_pid = pid; force_eid = eid;
@@ -2399,7 +2399,7 @@ main(
 	fprintf(pfa_file, "/UnderlinePosition %d def\n",
 		iscale(fontm.underline_position));
 
-	fprintf(pfa_file, "/UnderlineThickness %hd def\nend readonly def\n",
+	fprintf(pfa_file, "/UnderlineThickness %d def\nend readonly def\n",
 		iscale(fontm.underline_thickness));
 
 	fprintf(afm_file, "UnderlineThickness %d\n",

